package NYT.NProfiling.NProto;

option java_package = "ru.yandex.yt";
option java_multiple_files = true;

option python_package = "yt_proto.yt.core.profiling.proto";

enum EMetricType
{
    Counter = 0;
    Gauge = 1;
}

message TPoint
{
    required int64 time = 1;
    required int64 value = 2;
    repeated int64 tag_ids = 3;
    required EMetricType metric_type = 4;
    required string path = 5;
}

message TTag
{
    required int64 tag_id = 1;
    required string key = 2;
    required string value = 3;
}

message TPointBatch
{
    repeated TPoint points = 1;
    repeated TTag tags = 2;
}